smartsim/settingshold/batchSettings.py

```diff
@@ -0,0 +1,212 @@
+from __future__ import annotations
+import typing as t
+import copy
+
+from smartsim.log import get_logger
+from .._core.utils.helpers import fmt_dict
+from .common import process_env_vars, process_args, StringArgument
+from .batchCommand import SchedulerType
+from .translators.batch.pbs import QsubBatchArgTranslator
+from .translators.batch.slurm import SlurmBatchArgTranslator
+from .translators.batch.lsf import BsubBatchArgTranslator
+from .translators import BatchArgTranslator
+from .baseSettings import BaseSettings
+
+logger = get_logger(__name__)
+
+class BatchSettings(BaseSettings):
+    def __init__(
+        self,
+        scheduler: SchedulerType,
+        scheduler_args: t.Optional[t.Dict[str, t.Union[str,int,float,None]]] = None,
+        env_vars: t.Optional[StringArgument] = None,
+        **kwargs: t.Any,
+    ) -> None:
+        scheduler_to_translator = {
+            'sbatch' : SlurmBatchArgTranslator(),
+            'bsub' : BsubBatchArgTranslator(),
+            'qsub' : QsubBatchArgTranslator(),
+        }
+        if scheduler.value in scheduler_to_translator:
+            self.scheduler = scheduler
+        else:
+            raise ValueError(f"'{scheduler}' is not a valid scheduler name.")
+
+        if env_vars:
+            process_env_vars(env_vars)
+        self.env_vars = env_vars or {}
+
+        if scheduler_args:
+            process_args(scheduler_args)
+        self.scheduler_args = scheduler_args or {}
+
+        self.arg_translator = t.cast(BatchArgTranslator,scheduler_to_translator.get(scheduler.value))
```

Comment:
oh I see! Good suggestion, I am no longer searching through a dict to avoid creating each settings object and rather following the if/elif/else path, but I am keeping the 'from None' suggestion to suppress the traceback message!